ciao-controller/internal/datastore/datastore.go

```diff
@@ -1239,7 +1183,15 @@ func (ds *Datastore) GetInstanceLastStats(nodeID string) types.CiaoServersStats
 		if instance.NodeID != nodeID {
 			continue
 		}
-		serversStats.Servers = append(serversStats.Servers, instance)
+
+		i, err := ds.GetInstance(instance.ID)
+		if err != nil {
+			continue
```

Comment:
Is there an expected situation where this can fail (maybe stats message is received after instance deleted?) Should we at least log this error.